libyul/optimiser/ASTCopier.h

```diff
@@ -97,6 +97,19 @@ class ASTCopier: public ExpressionCopier, public StatementCopier
 	{
 		return _v ? std::make_shared<T>(translate(*_v)) : nullptr;
 	}
+
+	template <typename T>
+	std::unique_ptr<T> translate(std::unique_ptr<T> const& _v)
+	{
+		return _v ? std::make_unique<T>(translate(*_v)) : nullptr;
+	}
+
+	template <typename T>
+	boost::optional<T> translate(boost::optional<T> const& _v)
```

Comment:
This and the override for shared_ptr are not used and can be deleted.

Reply:
@chriseth Actually, even the shared_ptr<> version can be removed (dead code). so only the unique_ptr<> version remains.